### ISSUE TITLE
[Skeleton Tool] Use single parameter to define bone width

### ISSUE THREAD

**Issue by morevnaproject** [synfig/synfig#1557]
I have played a bit with current Skeleton Tool and I found that it will be more convenient to have one single parameter for bone width, instead of two separate parameters for origin and tip.

![screenshot_005](https://user-images.githubusercontent.com/332868/87326522-097c6880-c55d-11ea-8fca-e48b8567ea1a.png)

Can you change that, please? ^__^

**Comment by morevnaproject** [comment]
After thinking again on that, I got another idea:

It would be nice to have 2 parameters, but one will define width (tip and origin) for bones created in Regular Skeleton Layer, and second parameter will define width  (tip and origin) for Skeleton Deformation layer.

Also, it would be nice to temporary hide one parameter, depending on which layer type we are working with. ^__^

**Comment by Aa20475** [comment]
hey! Please test this in the latest commit!

**Comment by morevnaproject** [comment]
Hi! I have tested last commit. One correction:

> it would be nice to temporary hide one parameter, depending on which layer type we are working with

Currently it switches visible  parameter depending on layer type selected in Tool Options Panel (1).
But instead it is expected to switch visibility depending on active layer (2). ^__^

 
![screenshot_012](https://user-images.githubusercontent.com/332868/87678534-46d33700-c7a5-11ea-9ec5-251303c6f3f6.png)

**Comment by Aa20475** [comment]
Can you verify this? I tried something :sweat_smile:

**Comment by morevnaproject** [comment]
Hi! I have tested, but behavior is really weird... For example, I can't set Bone Width value to  0.10 or 1.50 - only integer values accepted. Also, at some moments it resets width to other weird values - like 0.0u or 7.0u...

**Comment by Aa20475** [comment]
Ohkay! Will fix it!

**Comment by morevnaproject** [comment]
I think you can implement it in the following way:

You can have two widgets in Tool Options Panel - one will define width (tip and origin) for bones created in Regular Skeleton Layer, and second parameter will define width (tip and origin) for Skeleton Deformation layer.

And all you have to do is just show/hide particular widget depending on the type of selected layer.

**Comment by Aa20475** [comment]
Alright! Will try that too! 😁

**Comment by Aa20475** [comment]
Hey! I've made some changes. I just am switching the sensitivity of the widget here. But, please let me know if the functionality is fine. Also, I had to change the `layer_change_handler`, so please verify issue #1603

**Comment by morevnaproject** [comment]
Hi! 
I like the idea with switching the sensitivity of the widget, it helps with debugging.

I have tested it and found the following issue:

1. Select Skeleton Tool and create a Skeleton Deformation Layer with several bones.
2. Switch to Arrow Tool. Make sure that Skeleton Deformation Layer is selected.
3. Select Skeleton Tool again -> The panel not updated (second widget should be sensitive, because Skeleton Deformation layer is active).

Also, I do not like the idea of mixing PR #1604 with issue #1603.

**Comment by morevnaproject** [comment]
> I have tested it and found the following issue:
> ...

Video - https://youtu.be/jj7fWfLVu0I

**Comment by morevnaproject** [comment]
...also, PR #1604 somehow introduces issues with bone parents:

1. When creating regular Skeleton Layer first bone is not assigned as a parent for second bone. For third bone and all subsequent everything is alright.
2. When creating Skeleton Deformation Layer bones are not parented at all.

Video - https://youtu.be/Hltme-gAhgQ

**Comment by Aa20475** [comment]
> Also, I do not like the idea of mixing PR #1604 with issue #1603.

Alright I'll split them up.

Will also try to fix the issues asap! ^__^

**Comment by Aa20475** [comment]
Hey! @morevnaproject did you test the recent commit?

**Comment by morevnaproject** [comment]
Hi! Testing it now...

**Comment by morevnaproject** [comment]
> But when clearing this issue, I think I'll have to play around with the `layer_change_handler`, so the other issue gets resolved with it?

Yes. So it's best to get  #1603 resolved first, and then do this issue. ;)

**Comment by Aa20475** [comment]
Okayy 😅😂

**Comment by morevnaproject** [comment]
Hi! I have tested and switching sensitivity for widgets now works well. I think you can make insensitive widgets hidden now. 

Regarding PR #1603 see my latest comments - 
https://github.com/synfig/synfig/pull/1618#issuecomment-678078071
